opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -346,8 +358,76 @@ class EclMaterialLawManager
 
             // read the end point scaling info for the saturation region
             unscaledEpsInfo_[satnumIdx].extractUnscaled(deck, eclState, satnumIdx);
+        }
+
+        EclEpsGridProperties epsGridProperties, epsImbGridProperties;
+        epsGridProperties.initFromDeck(deck, eclState, /*imbibition=*/false);
+        if (enableHysteresis())
+            epsImbGridProperties.initFromDeck(deck, eclState, /*imbibition=*/true);
+
+        initParamsForElements_(materialLawParams_, deck, eclState, compressedToCartesianElemIdx, satnumRegionArray,
+                               gasOilUnscaledPointsVector, oilWaterUnscaledPointsVector, gasOilEffectiveParamVector,oilWaterEffectiveParamVector,
+                               epsGridProperties, epsImbGridProperties );
+
 
+        // for well connections
+        std::vector<int> satnumRegionArrayConnections;
+        std::vector<int> compressedToCartesianElemIdxConnections;
+
+        const auto& schedule = eclState.getSchedule();
+        const auto& eclGrid = eclState.getInputGrid();
+
+        size_t timeStep = 0;
+        auto wells = schedule.getWells(timeStep); //need all wells ...
+        for (auto wellIter= wells.begin(); wellIter != wells.end(); ++wellIter) {
+            const auto* well = (*wellIter);
+            for(const auto& completion : well->getCompletions(timeStep)) { // ... and all connections!!!
+
+                int cartIdx = eclGrid.getGlobalIndex(completion.getI(), completion.getJ(), completion.getK());
+                compressedToCartesianElemIdxConnections.push_back(cartIdx);
+
+                int satnumId = completion.getSatTableId();
+                if (satnumId > 0)
+                    satnumRegionArrayConnections.push_back(satnumId);
+                else {
+                    int satnumIdFromCell = satnumRegionArray[eclGrid.activeIndex(cartIdx)];
+                    satnumRegionArrayConnections.push_back(satnumIdFromCell);
+                }
+            }
         }
+        // TODO change satnum/imbnum in epsGridProperties or change in ->extractScaled(...) ?
```

Comment:
probably none, i.e., use the unscaled values from the tables. Alternatively it could be the ones from the corresponding cell, but this would only make limited sense because the SATNUM/IMBNUM for the connection might be different from that of the cell...